File: PHP/level-500.php
<?php
    const FIRST_NUMBER = 1;
    const LAST_NUMBER = 100;

    for($i = FIRST_NUMBER; $i <= LAST_NUMBER; $i++) {
        echo to_fizz_buzz_string($i), "\n";
    }

    function to_fizz_buzz_string($number) {
        if (is_fizz_buzz($number)) {
            return 'FizzBuzz';
        } else if (is_fizz($number)) {
            return 'Fizz';
        } else if (is_buzz($number)) {
            return 'Buzz';
        } else {
            return (string) $number;
        }
    }

    function is_fizz_buzz($number) {
        return is_fizz($number) && is_buzz($number);
    }

    function is_fizz($number) {
        return $number % 3 == 0;
    }

    function is_buzz($number) {
        return $number % 5 == 0;
    }
?>

File: Java/level-500.java
public class Main {
    private static final int LOOP_COUNT = 100;

    public static void main(String[] args) {
        for (int i=1; i<=LOOP_COUNT; i++) {
            System.out.println(toFizzBuzzString(i));
        }
    }

    private static String toFizzBuzzString(int number) {
        if (isFizzBuzz(number)) {
            return "FizzBuzz";
        } else if (isFizz(number)) {
            return "Fizz";
        } else if (isBuzz(number)) {
            return "Buzz";
        } else {
            return String.valueOf(number);
        }
    }

    private static boolean isFizzBuzz(int number) {
        return isFizz(number) && isBuzz(number);
    }

    private static boolean isFizz(int number) {
        return number % 3 == 0;
    }

    private static boolean isBuzz(int number) {
        return number % 5 == 0;
    }
}
